libcloud/test/compute/test_libvirt_driver.py

```diff
@@ -65,103 +75,66 @@ def __init__(self):
-            'storageVolLookupByKey',
-            'storageVolLookupByPath',
-            'suspendForDuration',
-            'unregisterCloseCallback',
-            'virConnGetLastError',
+            'getCellsFreeMemory', 'getDomainCapabilities', 'getFreeMemory',
+            'getFreePages', 'getHostname', 'getInfo', 'getLibVersion',
+            'getMaxVcpus', 'getMemoryParameters', 'getMemoryStats',
+            'getSecurityModel', 'getSysinfo', 'getType', 'getURI',
+            'getVersion', 'interfaceDefineXML', 'interfaceLookupByMACString',
+            'interfaceLookupByName', 'isAlive', 'isEncrypted', 'isSecure',
+            'listAllDevices', 'listAllDomains', 'listAllInterfaces',
+            'listAllNWFilters', 'listAllNetworks', 'listAllSecrets',
+            'listAllStoragePools', 'listDefinedDomains',
+            'listDefinedInterfaces', 'listDefinedNetworks',
+            'listDefinedStoragePools', 'listDevices', 'listDomainsID',
+            'listInterfaces', 'listNWFilters', 'listNetworks', 'listSecrets',
+            'listStoragePools', 'lookupByID', 'lookupByName', 'lookupByUUID',
+            'lookupByUUIDString', 'networkCreateXML', 'networkDefineXML',
+            'networkEventDeregisterAny', 'networkEventRegisterAny',
+            'networkLookupByName', 'networkLookupByUUID',
+            'networkLookupByUUIDString', 'newStream', 'nodeDeviceCreateXML',
+            'nodeDeviceLookupByName', 'nodeDeviceLookupSCSIHostByWWN',
+            'numOfDefinedDomains', 'numOfDefinedInterfaces',
+            'numOfDefinedNetworks', 'numOfDefinedStoragePools', 'numOfDevices',
+            'numOfDomains', 'numOfInterfaces', 'numOfNWFilters',
+            'numOfNetworks', 'numOfSecrets', 'numOfStoragePools',
+            'nwfilterDefineXML', 'nwfilterLookupByName',
+            'nwfilterLookupByUUID', 'nwfilterLookupByUUIDString',
+            'registerCloseCallback', 'restore', 'restoreFlags',
+            'saveImageDefineXML', 'saveImageGetXMLDesc', 'secretDefineXML',
+            'secretLookupByUUID', 'secretLookupByUUIDString',
+            'secretLookupByUsage', 'setKeepAlive', 'setMemoryParameters',
+            'storagePoolCreateXML', 'storagePoolDefineXML',
+            'storagePoolLookupByName', 'storagePoolLookupByUUID',
+            'storagePoolLookupByUUIDString', 'storageVolLookupByKey',
+            'storageVolLookupByPath', 'suspendForDuration',
+            'unregisterCloseCallback', 'virConnGetLastError',
             'virConnResetLastError'
         ]
         for f in fnt:
-            self.__dict__[f] = stub
+            if f.startswith('is'):
+                self.__dict__[f] = yes
+            elif f.startswith('list'):
+                self.__dict__[f] = ary
+            else:
+                self.__dict__[f] = stub
 
 
 class LibvirtNodeDriverTestCase(LibvirtNodeDriver, unittest.TestCase):
+    type = Provider.LIBVIRT
+    name = 'Libvirt'
+    website = 'http://libvirt.org/'
+
+    NODE_STATE_MAP = {
```

Comment:
Hm, do we need this attributes here on the test class? They are already defined in the `LibvirtNodeDriver` class.

Also, usually the idea is not to have the test class inherit from the Driver class, but the test case should instantiate the driver class and use that.

For example:

```python
def setUp(self):
    ....
    self.driver = LibvirtNodeDriver(...)

def some_test_method(self):
    nodes = self.driver.list_nodes()
    ...
```